src/components/Comments.js

```diff
@@ -0,0 +1,272 @@
+// @flow
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Modal, Button, Col, Row, Glyphicon } from 'react-bootstrap';
+
+import QueryResponse from '../api/QueryResponse';
+import AuthUtils from '../util/AuthUtils';
+import SimpleQueryRequest from '../api/SimpleQueryRequest';
+
+type CommentsProps = {
+  /** The document's ID, needed for adding/removing comments */
+  docId: string,
+  /** Optional. The table name to be used for comments document stored in the index */
+  commentsTable?: string,
+}
+
+type CommentsDefaultProps = {
+  commentsTable: string,
+}
+
+type CommentsState = {
+  comment: string;
+  response: QueryResponse | null;
+  error: string | null;
+  showCommentModal: boolean,
+  commentList: Array<any>,
+}
+
+class Comments extends React.Component<CommentsDefaultProps, CommentsProps, CommentsState> {
+  static contextTypes = {
+    searcher: PropTypes.any,
+  };
+
+  static defaultProps = {
+    commentsTable: 'comments',
+  };
+
+  static displayName = 'Comments';
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      comment: '',
+      response: null,
+      error: null,
+      showCommentModal: false,
+      commentList: [],
+    };
+    (this: any).showCommentModal = this.showCommentModal.bind(this);
+    (this: any).hideCommentModal = this.hideCommentModal.bind(this);
+    (this: any).captureComment = this.captureComment.bind(this);
+    (this: any).saveComment = this.saveComment.bind(this);
+    (this: any).clearComment = this.clearComment.bind(this);
+    (this: any).getComments = this.getComments.bind(this);
+    (this: any).deleteComment = this.deleteComment.bind(this);
+  }
+
+  componentDidMount() {
+    this.getComments();
+  }
+
+  getComments() {
+    const { searcher } = this.context;
+    const { docId, commentsTable } = this.props;
+    if (searcher) {
+      const qr = new SimpleQueryRequest();
+      qr.query = `AND(table:${commentsTable}, docId_s:FACET("${docId}"))`;
+      qr.facets = [];
+      qr.queryLanguage = 'advanced';
+      searcher.doCustomSearch(qr, (response: QueryResponse | null, error: string | null) => {
+        if (response) {
+          this.setState(
+            {
+              response,
+              comment: '',
+            },
+            this.populateComments,
+          );
+        } else if (error) {
+          this.setState({
+            error,
+          });
+        }
+      });
+    }
+  }
+
+  deleteComment(id: string) {
+    const jsonDoc = `{ "id" : "${id}", "mode" : "DELETE" }`;
+    this.context.searcher.search.addOrDeleteDocument(JSON.parse(jsonDoc), this.getComments);
+  }
+
+  populateComments() {
+    const { response } = this.state;
+    const loggedInUser = AuthUtils.getUserName(AuthUtils.getSavedUser());
+    if (response) {
+      const commentList = [];
+      response.documents.forEach((doc) => {
```

Comment:
The response formatting should be done in a generic location. We follow a pattern of saving requests in the `api` directory - as you did - and also creating a js class created from the result. `api/SimplePrincipal` is an example of this. If you do that on line 71 so that instead of storing `response` in state directly, you're storing `commentList`, which would be: `Array<Comment>`, then here you could just map the result and `Comment` would be it's own `.js` file and class that could potentially be used elsewhere.